Cython/Compiler/Optimize.py

```diff
@@ -4424,6 +4424,8 @@ def _multiply_string(self, node, string_node, multiplier_node):
                 string_node.unicode_value = encoded_string(
                     string_node.unicode_value * multiplier,
                     string_node.unicode_value.encoding)
+            if not string_node.value.is_unicode:
+                build_string = bytes_literal
```

Comment:
The constant folding [prefers](https://github.com/cython/cython/blob/75da5a9e1996830b83d5605f85caba2a9fde4690/Cython/Compiler/ExprNodes.py#L1732-L1735) the unicode value over the bytes value in the case of a `StringNode`, so we should do that here as well and assign `string_node.unicode_value` to `string_node.constant_result` below (instead of `string_node.value`).

Reply:
Also, this is probably best spelled more generally as `build_string = encoded_string if string_node.value.is_unicode else bytes_literal` above, since that applies to all cases.